lfsapi/ssh.go

```diff
@@ -81,7 +81,7 @@ func (c *sshAuthClient) Resolve(e Endpoint, method string) (sshAuthResponse, err
 		return res, nil
 	}
 
-	exe, args := sshGetLFSExeAndArgs(c.os, e, method)
+	exe, args := sshGetLFSExeAndArgs(c.os, c.git, e, method)
```

Comment:
I wonder if we could teach this function to accept instead a `*config.Configuration` as part of a future change. The type `*config.Configuration` has both a `.OS` and `.Git` environment attached to it, so I think that it might make more sense to pass that as one argument than these two.

I took a quick look through the code, and I think that this is better suited as a future change (instead of one in this pull request) since the callers run pretty deep. From my cursory look, I think we'd need to:

1. Change the signature of `lfsapi.Config` to instead have a `Config() *config.Configuration`.
2. Change the callers of `lfsapi.NewConfig()`. Potentially we could remove this function (and type) entirely, and instead use the [options pattern](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) (a good example is [here](https://github.com/git-lfs/git-lfs/blob/v2.5.1/git/githistory/rewriter.go#L149-L173) and [here](https://github.com/git-lfs/git-lfs/blob/v2.5.1/git/githistory/rewriter.go#L196-L198)).
3. Make the rest of the changes.

It also might be worth _not_ doing this, since I think that there is a version of `NewContext` that takes two `map[string]string` and emulates an OS/Git configuration environment. It might be worth exposing something similar from the `config` package and using that, since I think that this facility is useful in places other than package `lfsapi`.

Reply:
I think that would be a nice improvement, but I agree that would be better in a different pull request.  I thought about doing that in this change, but didn't because it would (as you mention) involve a significant refactor that seemed like more than I wanted to do here.

I also think it might conflict with some work to refactor the HTTP client that @PastelMobileSuit is doing, so deferring that change until after his code lands would probably be more courteous.

Reply:
Sounds good. Let's wait for the changes that @PastelMobileSuit is working and merge this whenever you are ready to do so.